modules/framework/galasa-parent/dev.galasa.framework.k8s.controller/src/main/java/dev/galasa/framework/k8s/controller/TestPodScheduler.java

```diff
@@ -261,6 +262,40 @@ V1Pod createTestPod(String runName, String engineName, boolean isTraceEnabled) {
             }
         }
 
+        String nodeTolerations = this.settings.getNodeTolerations();
```

Comment:
can this pls be put into a separate method, as it's getting pretty big...

Reply:
Also, any unit tests for that method would be good ?

Reply:
Hi, moved the nodetoleration creation into it's own method. I've also added a unit test for the nodetolerations creation. I spotted that the nodeaffinity creation didn't have a unit test already, so I've added one for that too.